x-pack/apm-server/aggregation/txmetrics/aggregator.go

```diff
@@ -594,3 +644,35 @@ func transactionCount(tx *model.Transaction) float64 {
 	}
 	return 1
 }
+
+func writeLabels(w io.StringWriter, l model.Labels) {
+	if len(l) == 0 {
+		return
+	}
+	for key, value := range l {
+		w.WriteString(key)
+		if value.Value != "" {
+			w.WriteString(value.Value)
+			continue
+		}
+		for _, v := range value.Values {
+			w.WriteString(v)
+		}
+	}
+}
+
+func writeNumericLabels(w io.StringWriter, l model.NumericLabels) {
+	if len(l) == 0 {
+		return
+	}
+	for key, value := range l {
+		w.WriteString(key)
+		if value.Value != 0 {
+			w.WriteString(strconv.FormatFloat(value.Value, 'e', -1, 64))
```

Comment:
Can we use encoding/binary.LittleEndian.PutUint64 & math.Float64bits? Should be faster and create less garbage I think?